sdcm/cluster.py

```diff
@@ -3771,35 +3774,22 @@ def critical_node_setup_events():
             add_severity_limit_rules([f'{e["rule_pattern"]}={e["default_severity"]}' for e in critical_events])
 
         start_time = time.perf_counter()
-        init_nodes = []
-        results = []
 
         if isinstance(cl_inst, BaseScyllaCluster):
             # Update installed scylla before node setup, scylla server will be start at the end of node_setup
             cl_inst.update_db_binary(node_list, start_service=False)
             cl_inst.update_db_packages(node_list, start_service=False)
 
         with critical_node_setup_events():
+            start_time = time.perf_counter()
+            cl_inst.run_func_parallel(func=node_setup, node_list=node_list)
             for node in node_list:
-                if isinstance(cl_inst, BaseScyllaCluster) \
-                        and not getattr(cl_inst, 'params', {}).get('use_legacy_cluster_init'):
-                    init_nodes.append(node)
-                    start_time = time.perf_counter()
-                    node_setup(node)
-                    verify_node_setup(start_time)
-                else:
-                    setup_thread = threading.Thread(target=node_setup, name='NodeSetupThread',
-                                                    args=(node,), daemon=True)
-                    setup_thread.start()
-                    if isinstance(cl_inst, BaseScyllaCluster):
-                        cl_inst.log.info("Wait 120 seconds before next node setup")
-                        time.sleep(120)
-
-            while len(results) != len(node_list):
-                verify_node_setup(start_time)
-
+                node_startup(node, startup_queue)
```

Comment:
when using raft, maybe we could do it in parallel?

Reply:
It is out of scope for this change. As well as setting up loader nodes in parallel to DB nodes.
This change makes the `setup` be done in parallel, not `startup`.

Also, in raft we cannot startup all at once, we can boot more than 1 having alive cluster.
For example, Boot up 2 new nodes having cluster of 3 existing alive nodes.
@aleksbykov right?

Reply:
> It is out of scope for this change. As well as setting up loader nodes in parallel to DB nodes.
> This change makes the `setup` be done in parallel, not `startup`.

yes, setup could be done in parallel. Loader was started after db_nodes, to avoid not needed loader setup startup, if db nodes are failed to start


> Also, in raft we cannot startup all at once, we can boot more than 1 having alive cluster.
> For example, Boot up 2 new nodes having cluster of 3 existing alive nodes.
> @aleksbykov right?

It was related not to raft, but to gossiper. With new feature raft-topology which will be in 6.0, node could be started in parallel